packages/taxdat/R/setup_helpers.R

```diff
@@ -1,3 +1,164 @@
+#' @export
+check_update_config <- function(config_fname, covariate_list = NULL){
+  
+  ### Read in the config file first 
+  cholera_testing <- as.logical(Sys.getenv("CHOLERA_TESTING", "FALSE"))
+  config_file <- yaml::read_yaml(config_fname)
+
+  ### If no file exists to tell us what covariates exist, here's a hard coded list
+  if(is.null(covariate_list)){
+    all_covariates <- c("", "pop", "dist_to_water",              
+                        "dist_to_lakes", "dist_to_rivers",              
+                        "dist_to_coast", "san_access",                  
+                        "water_access", "open_defe",            
+                        "stunting", "wasting", 
+                        "access_cities", "access_healthcare_motorized", 
+                        "access_healthcare_walking")
+  }else{
+    all_covariates <- names(yaml::read_yaml(covariate_list)) #tmp default: "Layers/middle-distance_covariates.yml"
+  }
+
+  ### Make the check list 
+  check_list <- list(
+    name = "no-check", 
+    countries = "no-check", 
+    countries_name = "no-check", 
+    country_data_report_filename = function(nm, cty, ctynm, param){
+                if(is.null(param)){return(paste(nm, cty, ctynm, "country-data-report.html", sep="_"))}
+                return(param)
+              }, 
+    aoi = function(param){
+        if(cholera_testing){return(param)}
+        return("raw")
+      }, 
+    res_space = "no-check", 
+    res_time = as.function(check_time_res), 
+    grid_rand_effects_N = as.function(check_grid_rand_effects_N), 
+    case_definition = as.function(check_case_definition), 
+    start_time = function(param){
+                if(is.null(param)){stop("The start_time parameter should not be blank because there is no default")}
+                return(param)
+              },   
+    end_time = function(param){
+              if(is.null(param)){stop("The end_time parameter should not be blank because there is no default")}
+              return(param)
+            }, 
+    data_source = function(param){
+                if(is.null(param)){return("sql")}
+                return(param)
+              }, 
+    ovrt_metadata_table = function(param){
+                        if(is.null(param)){return("no")}
+                        return(param)
+                      }, 
+    OCs = "no-check", 
+    taxonomy = function(param){
+              if(is.null(param)){return("taxonomy-working/working-entry1")}
+              return(param)
+            },  
+    covariate_choices = as.function(check_covariate_choices), 
+    obs_model = function(param){
+              if(is.null(param)){return(1)}
+              if(!as.numeric(param) %in% 1:3){return(1)}
+              return(as.numeric(param))
+            }, 
+    od_param = function(param1, param2){
+              if(as.numeric(param1) == 2 | as.numeric(param1) == 3){return(as.numeric(param2))}
+              return(NULL)
+            }, 
+    time_effect = "stan-check", 
```

Comment:
So, all the values with Stan-check get overwritten in line 147 when the get_stan_parameters output is appended?